src/main/java/millom/sandbox/service/MyClientService.java

```diff
@@ -0,0 +1,48 @@
+package millom.sandbox.service;
+
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.ProcessingException;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.core.MediaType;
+import millom.sandbox.CustomException.InvalidWeatherException;
+
+public class MyClientService {
+
+  private static final String REST_BASE_URI
+      = "https://mars.nasa.gov/rss/api/";
+  private static final String API_QUERY_PARAMS = "?feed=%s&feedtype=%s&ver=%.1f&category=%s";
+  private final Client client;
+
+  public MyClientService(Client client) {
```

Comment:
The `client` instance can also be injected from `MainApp`.

Reply:
Injected `client` instance from `MainApp. `

```
@Inject
  private Client client;
```
In the MainApp.java

`final Client client = ClientBuilder.newClient();`

` bind(client).to(Client.class);`